lib/Rex/Pkg/ArchLinux.pm

```diff
@@ -0,0 +1,100 @@
+#
+# (c) Harm Müller <harm _DOT_ mueller _AT_ g m a i l _Dot_ com>
+#
+# vim: set ts=2 sw=2 tw=0:
+# vim: set expandtab:
+
+package Rex::Pkg::ArchLinux;
+
+use strict;
+use warnings;
+
+# VERSION
+
+use Rex::Commands::Run;
+use Rex::Helper::Run;
+use Rex::Commands::File;
+use Rex::Commands::Fs;
+
+use Rex::Pkg::Base;
+use base qw(Rex::Pkg::Base);
+
+sub new {
+  my $that  = shift;
+  my $proto = ref($that) || $that;
+  my $self  = $proto->SUPER::new(@_);
+
+  bless( $self, $proto );
+
+  $self->{commands} = {
+    install           => 'pacman --noprogressbar --noconfirm --needed -S %s',
+    install_version   => 'pacman --noprogressbar --noconfirm --needed -S %s',     # makes no sense to specify the package version
+    update_system     => 'pacman --noprogressbar --noconfirm -Syu',
+    remove            => 'pacman --noprogressbar --noconfirm -Rs %s',
+    purge             => 'pacman --noprogressbar --noconfirm -Rns %s',
+    update_package_db => 'pacman --noprogressbar -Sy',
+  };
+
+  return $self;
+}
+
+sub bulk_install {
+  my ( $self, $packages_aref, $option ) = @_;
+
+  delete $option->{version}; # makes no sense to specify the same version for several packages
+
+  $self->update( "@{$packages_aref}", $option );
+
+  return 1;
+}
+
+sub get_installed {
+  my ( $self, $pkg ) = @_;
+  my @pkgs;
+  my $pkg_query = 'pacman -Qq';
+  if ($pkg) {
+    $pkg_query .= " " . $pkg;
+  }
+  my $pkg_query_details = "pacman -Qi";
+
+  my @installed_packages = i_run $pkg_query;
+  for my $package (@installed_packages) {
+    my @lines = i_run $pkg_query_details;
+    my ($tmp_name, $tmp_version, $tmp_release, $tmp_arch);
+    for my $line (@lines) {
+      if ( $line =~ m/^Name\s+\:\s+([^\s]+)$/ ) {
+        $tmp_name = $1;
+      }
+      if ( $line =~ m/^Version\s+\:\s+([^\s]+)-([^\s]+)$/ ) {
+        $tmp_version = $1;
+        $tmp_release = $2;
+      }
+      if ( $line =~ m/^Architecture\s+\:\s+([^\s]+)$/ ) {
+        $tmp_arch = $1;
+      }
+    }
+    push(
+      @pkgs,
+      {
+        name    => $tmp_name,
+        version => $tmp_version,
+        release => $tmp_release,
+        arch    => $tmp_arch,
+      }
+    );
+  }
+
+  return @pkgs;
+}
+
+sub add_repository {
+  #no suitable repo management use template/file for pacman.conf
```

Comment:
Would it make sense to print that message out as a warning, instead of only having it as a code comment, and silently returning success? Same question applies below in `rm_repository` too.